### ISSUE TITLE
Looping Sounds?

### ISSUE THREAD

**Issue by AbuMaia** [armonge/gnome-shell-extension-focusli#4]
Are the sounds supposed to loop until they are turned off? I am not seeing this behaviour on my 17.04. The sounds play for a little while, then quit, and I have to turn them off and back on again to get them to play again.

**Comment by lachmanfrantisek** [comment]
Restarting gnome shell solves the problem for me. Don't know where is the problem, but try logging out and login in.

**Comment by felipeborges** [comment]
Yes, this bug is caused because in the first run the files are streamed over the internet. In the next shell reboot it loads them locally. So any bandwidth limitation/interruption in the first run might cause the extension not to work properly. I will fix it asap.

**Comment by AbuMaia** [comment]
Restarting Gnome Shell didn't help. Rebooting the computer didn't help. Sounds still play for a little bit, then stop with their button still active.

**Comment by obguy** [comment]
Restarting gnome shell worked for me.

**Comment by AbuMaia** [comment]
I have restarted Gnome Shell, rebooted the computer, uninstalled the extension, restarted GS, reinstalled the extension, and restarted GS again, and it still does not loop the sounds. They play once, then stop. Unlike the user with the installation trouble, the sounds do not start up again on their own after some time.

**Comment by piratenpanda** [comment]
At least the end and beginning of the rain sound is very noticeable. Would be nice if the sounds would really loop without any interruption. Should i open a new bug for that?

**Comment by piratenpanda** [comment]
Also when using multiple sound at the same time some just stop without looping

**Comment by AbuMaia** [comment]
It seems to me how the sounds were produced is not a responsibility of this extension, being accessed from a third party site, so should not be a bug of this extension.

**Comment by felipeborges** [comment]
Yes, I see that some people are experiencing the "no loop" bug. I am investigating a fix for the next version.

About the sounds, they will be updated in the next release to better sounds.

**Comment by AbuMaia** [comment]
Ok, now after I installed another extension, I'm getting the Play--Stop--Play--Stop phenomenon. It is looping, but there's a 19 second gap when playing the Rain sound. This happens with all sounds, and the gap for each sound is the same length as the sound itself.

**Comment by AbuMaia** [comment]
Oh well, I'm giving up on this extension and removing it until it gets fixed. No sense having it take up space and memory on my computer when it doesn't work.

**Comment by felipeborges** [comment]
@AbuMaia well, if there is still the bug for you. The Issue is not resolved.

Please, bear in mind that I do this in my free time, and as the license states, "This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details".

**Comment by alourie** [comment]
@felipeborges anything new on this by any chance? Just found this extension, it is exactly what I need but sounds still do not loop. Any direction here?

Cheers.

**Comment by felipeborges** [comment]
@alourie could you please try whether commit 88df5eb2 fixed it for you?

**Comment by AbuMaia** [comment]
Just reinstalled it to try again. Sounds still stop, just as described in the original comment. Reinstalled through the Gnome extensions page, so I don't know if the recent commit is included or not.

**Comment by AbuMaia** [comment]
Just reinstalled it to try again. Sounds still stop, just as described in the original comment. Reinstalled through the Gnome extensions page, so I don't know if the recent commit is included or not.
